### DISCUSSION TITLE
Anchor

### DISCUSSION THREAD

**Discussion by DFergFLA** [XusinboyBekchanov/VisualFBEditor#568] [Q&A] [answered]
So...  I'm back playing with VFBEditor again.  I think "Anchor" is what I'm trying to work out.  Still a newbie with very, very little experience.  I am trying to be able setup some controls to "resize" as the form gets bigger/smaller.  I've tried using the "Anchor"  property but either I am not understanding what this is or I am using it wrong.  Not sure which.  Either way I clearly do not have enough information on how to use it.  So.. hit me!  

Also, is there anything that can be used as a "manual" or guide to VFPEditor?   I don't want to bombard the message board with a bazillion small questions if there is a reference out there. 

Thanks

**Comment by XusinboyBekchanov** [comment]
There is no manual, but the anchor is used in this way:

```
'#Region "Form"
	#if defined(__FB_MAIN__) AndAlso Not defined(__MAIN_FILE__)
		#define __MAIN_FILE__
		#ifdef __FB_WIN32__
			#cmdline "Form1.rc"
		#endif
		Const _MAIN_FILE_ = __FILE__
	#endif
	#include once "mff/Form.bi"
	#include once "mff/CommandButton.bi"
	
	Using My.Sys.Forms
	
	Type Form1Type Extends Form
		Declare Constructor
		
		Dim As CommandButton CommandButton1
	End Type
	
	Constructor Form1Type
		' Form1
		With This
			.Name = "Form1"
			.Text = "Form1"
			.Designer = @This
			.SetBounds 0, 0, 350, 300
		End With
		' CommandButton1
		With CommandButton1
			.Name = "CommandButton1"
			.Text = "CommandButton1"
			.TabIndex = 0
			.Anchor.Top = AnchorStyle.asAnchor
			.Anchor.Left = AnchorStyle.asAnchor
			.Anchor.Right = AnchorStyle.asAnchor
			.Anchor.Bottom = AnchorStyle.asAnchor
			.SetBounds 20, 40, 290, 50
			.Designer = @This
			.Parent = @This
		End With
	End Constructor
	
	Dim Shared Form1 As Form1Type
	
	#if _MAIN_FILE_ = __FILE__
		App.DarkMode = True
		Form1.MainForm = True
		Form1.Show
		App.Run
	#endif
'#End Region
```

If you need it proportionally, then like this:

```
'#Region "Form"
	#if defined(__FB_MAIN__) AndAlso Not defined(__MAIN_FILE__)
		#define __MAIN_FILE__
		#ifdef __FB_WIN32__
			#cmdline "Form1.rc"
		#endif
		Const _MAIN_FILE_ = __FILE__
	#endif
	#include once "mff/Form.bi"
	#include once "mff/CommandButton.bi"
	
	Using My.Sys.Forms
	
	Type Form1Type Extends Form
		Declare Constructor
		
		Dim As CommandButton CommandButton1, CommandButton2
	End Type
	
	Constructor Form1Type
		' Form1
		With This
			.Name = "Form1"
			.Text = "Form1"
			.Designer = @This
			.SetBounds 0, 0, 350, 300
		End With
		' CommandButton1
		With CommandButton1
			.Name = "CommandButton1"
			.Text = "CommandButton1"
			.TabIndex = 0
			.Anchor.Top = AnchorStyle.asAnchor
			.Anchor.Left = AnchorStyle.asAnchor
			.Anchor.Right = AnchorStyle.asAnchorProportional
			.Anchor.Bottom = AnchorStyle.asAnchor
			.SetBounds 20, 40, 140, 50
			.Designer = @This
			.Parent = @This
		End With
		' CommandButton2
		With CommandButton2
			.Name = "CommandButton2"
			.Text = "CommandButton2"
			.TabIndex = 1
			.Anchor.Top = AnchorStyle.asAnchor
			.Anchor.Left = AnchorStyle.asAnchorProportional
			.Anchor.Right = AnchorStyle.asAnchor
			.Anchor.Bottom = AnchorStyle.asAnchor
			.SetBounds 170, 40, 150, 50
			.Designer = @This
			.Parent = @This
		End With
	End Constructor
	
	Dim Shared Form1 As Form1Type
	
	#if _MAIN_FILE_ = __FILE__
		App.DarkMode = True
		Form1.MainForm = True
		Form1.Show
		App.Run
	#endif
'#End Region
```

**Comment by DFergFLA** [comment]
Thank you.  Can I assume those properties can be set using the properties tab.  

Would there be any benefit to me getting a VB6 book?  How close are these things following VB6?   I know it wouldn't be perfect but it may be something.

**Comment by XusinboyBekchanov** [comment, marked as answer]
>Thank you. Can I assume those properties can be set using the properties tab.

Yes, I set this too from the properties window:
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/32607344/d002e314-ecc1-4d9f-8290-fde837fabaac)

>Would there be any benefit to me getting a VB6 book? How close are these things following VB6? I know it wouldn't be perfect but it may be something.

You can look at VB6 and VB.Net. The Anchor property is available in VB.Net, but it’s a little different.

**Reply by DFergFLA** [reply]
Thank you again.  I will probably pick up a VB6 book to use as starting point for looking up things.  If that fails I will be here.    I used VB5 and 6 back in the 90's but have not really touched any programming languages much since.  I was never dev, just a nerd.
